src/core/api_client/apiClient.ts

```diff
@@ -42,7 +42,7 @@ export class ApiClient {
   private refreshToken = '';
   private tokenWillExpire: Date | undefined;
   private checkTokenPromise: Promise<boolean> | undefined;
-  private expiredTokenCallback?: () => void;
+  public expiredTokenCallback?: () => void;
```

Comment:
I would refactor it a bit: 

1 token expiration is responsibility of apiClient, so authClient should know nothing about expiredTokenCallback
so in AuthClient.ts line 111
`const response = this._apiClient.getLocalAuthToken(this.logout);`

and remove onTokenExpired

2 in apiClient.ts 

public expiredTokenCallback -> public authClientLogoutCallback 
because it's not only about token  expiration

add
```
  private onTokenExpired() {
    if (this.authClientLogoutCallback) {
      console.warn('User session has been expired. Logging out.');
      this.authClientLogoutCallback();
    }
  }
```
  
  and would call it everywhere instead of 
  
```
  if (this.expiredTokenCallback) {
    this.expiredTokenCallback();
  }
```
  
  so in the end in apiClientInstance.ts
  
  apiClient.expiredTokenCallback?.(); -> apiClient.authClientLogoutCallback?.();